File: windows/DMShot/Capture/CaptureGeometry.cs
using System.Drawing;
namespace DMShot.Capture;

/// <summary>
/// Pure geometry for the capture → in-place overlay handoff. Converts a selection
/// rect expressed in a display's local pixel space into a global physical-screen
/// pixel rect. Windows is top-left origin (no Y-flip), so we only add the display
/// origin. Mirrors macOS CaptureGeometry.screenRect (which flips Y for AppKit).
/// </summary>
public static class CaptureGeometry
{
    public static PixelRect ScreenRect(PixelRect selectionInDisplay, Rectangle displayBoundsPx)
        => new(displayBoundsPx.Left + selectionInDisplay.X,
               displayBoundsPx.Top + selectionInDisplay.Y,
               selectionInDisplay.Width,
               selectionInDisplay.Height);
}

File: mac/Sources/DMShot/CaptureGeometry.swift
import CoreGraphics

/// Pure geometry for the capture → in-place overlay handoff.
enum CaptureGeometry {
    /// Convert a selection rect expressed in a display's **local, top-left-origin**
    /// point space into a **global AppKit screen rect** (bottom-left origin).
    /// `displayFrameGlobal` is the display's frame in global screen points.
    static func screenRect(selection: CGRect, in displayFrameGlobal: CGRect) -> CGRect {
        CGRect(
            x: displayFrameGlobal.minX + selection.minX,
            y: displayFrameGlobal.maxY - selection.maxY,
            width: selection.width,
            height: selection.height)
    }
}
